File: backend/pokerogue/src/main/java/com/pokerogue/helper/ability/dto/AbilityResponse.java
package com.pokerogue.helper.ability.dto;

import com.pokerogue.helper.ability.data.Ability;

public record AbilityResponse(String id, String koName, String description) {

    public static AbilityResponse from(Ability ability) {
        return new AbilityResponse(ability.getId(), ability.getKoName(), ability.getDescription());
    }
}

File: android/remote/src/main/java/poke/rogue/helper/remote/dto/response/ability/AbilityResponse.kt
package poke.rogue.helper.remote.dto.response.ability

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AbilityResponse(
    val id: String,
    @SerialName("koName")
    val name: String,
    val description: String,
)
